### PR TITLE
cpu/kinetis/i2c: Fix false positive for expected EIO during i2c write

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
This fixes the positive result when master write data is NACKed.
This false positive occurs when the write frame is finished but a data nack occurred.
The AF check should occur first.

Found with the HiL CI.

### Contribution description
Check if EIO error happened before checking if the frame is finished.

### Testing procedure

Simulate a data nack (you can use philip)
Write i2c byte and it should be giving an -EIO but is says everything is fine (in master), we should get an EIO with this PR.

### CODE REVIEW

**Comment by smlng** [comment]
please add cppcheck fix as separate commit

**Comment by MrKevinWeiss** [comment]
Arg, it is actually being used though.  I think that is what the `++dummy` is supposed to do.  I also don't get the error on my local machine when running cppcheck.

**Comment by benemorius** [comment]
> I think that is what the `++dummy` is supposed to do

Either that or it's ensuring that the register read doesn't get optimized out. Does `(void)dummy` work? I don't know whether that works to prevent overzealous optimization though.

File: cpu/kinetis/periph/i2c.c [hunk 1]
```diff
@@ -379,6 +379,8 @@ int i2c_read_bytes(i2c_t dev, uint16_t addr, void *data, size_t len, uint8_t fla
         bit_set8(&i2c->C1, I2C_C1_IICIE_SHIFT);
         /* Initiate master receive mode by reading the data register once when
          * the C1[TX] bit is cleared and C1[MST] is set */
+         /* cppcheck-suppress unreadVariable
+          * (reason: needed to empty D register) */
         volatile uint8_t dummy;
         dummy = i2c->D;
```
Comment (smlng) [inline cpu/kinetis/periph/i2c.c]: the suppression needs to go above this line, not the variable declaration

**Comment by smlng** [comment]
yeah dummy is being used (i.e. changed) but not never **read** and that's what cppcheck is complaining about, bc normally why would you change a variable (which costs cpu cycles) when its new value is never used, so you could just omit changing the value in the first place.

**Comment by MrKevinWeiss** [comment]
Ahh, good testing locally and it seems to fix it.  Now we wait for the CI

File: cpu/kinetis/periph/i2c.c [hunk 2]
```diff
@@ -380,8 +380,9 @@ int i2c_read_bytes(i2c_t dev, uint16_t addr, void *data, size_t len, uint8_t fla
         /* Initiate master receive mode by reading the data register once when
          * the C1[TX] bit is cleared and C1[MST] is set */
         volatile uint8_t dummy;
+        /* cppcheck-suppress unreadVariable
+         * (reason: needed to empty D register) */
         dummy = i2c->D;
-        ++dummy;
```
Comment (smlng) [inline cpu/kinetis/periph/i2c.c]: why did you remove that, now the compiler complains 😬

**Comment by MrKevinWeiss** [comment]
@smlng I think she is ready!
